reactioncommerce/reaction
# Audax fix issue 2750

Resolves #2750 

So on the `client` side when a shop is being fetched from the `Shops` collection by `id` it sometimes caused the issue where at times fetching a shop throws `undefined`. I resolved this by subscribing to the shops collection of types `PrimaryShop` and `MerchantShop` , before fetching for a shop from the `Shops` collection.

##### HOW TO TEST 
- Login as admin
- Create a second shop
- As marketplace admin switch to second shop
- Observe that there are no more console errors like below
```
TypeError: Cannot read property 'brandAssets' of undefined
    at Brand.getLogo (brand.js:23)
    at Brand.render (brand.js:33)
    at modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107773
    at measureLifeCyclePerf (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107053)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107772)
    at ReactCompositeComponentWrapper._renderValidatedComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107799)
    at ReactCompositeComponentWrapper._updateRenderedComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107723)
    at ReactCompositeComponentWrapper._performComponentUpdate (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107701)
    at ReactCompositeComponentWrapper.updateComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107622)
    at ReactCompositeComponentWrapper.receiveComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107524)
```
